lib/index.ts

```diff
@@ -726,3 +729,31 @@ function toSnykVersion(v: ModuleVersion): string {
     throw new Error('Unexpected module version format');
   }
 }
+
+function constructPurl(mod: GoModule, importPath: string): string | undefined {
+  let namespace: string | undefined;
+  let name: string | undefined;
+  const version = mod.Version;
+  let subpath: undefined | string;
+
+  const idx = mod.Path.lastIndexOf('/');
+  if (idx < 0) {
+    name = mod.Path;
+  } else {
+    namespace = mod.Path.slice(0, idx);
+    name = mod.Path.slice(idx + 1);
+  }
```

Comment:
A comment or two could be nice. I was thinking like an example with actual string and what happens.